test/add_key/processor-spec.ts

```diff
@@ -0,0 +1,704 @@
+    });
+
+    it('should throw and error if array cannot be converted to numbers', async () => {
+        const test = await makeTest({
+            key_fields: [
+                'name',
+                'age',
+                'location'
+            ],
+            truncate_location: [
+                'location'
+            ],
+            truncate_location_places: 4
+        });
+
+        const data = cloneDeep(testData).map((doc) => {
+            doc.location = ['badPoint', '55.3454349123934'];
+            return doc;
+        });
+
+        await expect(test.runSlice(data)).rejects.toThrow();
+    });
+
+    it('should throw an error if object geo-point cannot be truncated', async () => {
+        const test = await makeTest({
+            key_fields: [
+                'name',
+                'age',
+                'location'
+            ],
+            truncate_location: [
+                'location'
+            ],
+            truncate_location_places: 4
+        });
+
+        const data = cloneDeep(testData).map((doc) => {
+            doc.location = { lon: 'no number here', lat: 55.3454349123934 };
+            return doc;
+        });
+
+        await expect(test.runSlice(data)).rejects.toThrow();
+    });
+
+    it('should not add empty fields or objects to key', async () => {
+        const test = await makeTest();
+
+        const data = cloneDeep(testData);
+
+        data[0].foo = [];
+        data[1].bar = {};
+        data[2].foo = null;
+        data[2].bar = false;
+
+        const results = await test.runSlice(data);
+
+        expect(results).toEqual([
+            {
+                name: 'bob',
+                age: 122,
+                foo: [],
+                _key: makeKey([122, 'bob'], 'md5')
+            },
+            {
+                name: 'joe',
+                age: 34,
+                bar: {},
+                _key: makeKey([34, 'joe'], 'md5')
+            },
+            {
+                name: 'frank',
+                age: 99,
+                foo: null,
+                bar: false,
+                _key: makeKey([99, false, 'frank'], 'md5')
+            }
+        ]);
+    });
+
+    it('should correctly handle data windows', async () => {
+        const test = await makeTest();
+
+        const dataWindow = DataWindow.make(1, testData);
+
+        const results = await test.runSlice([dataWindow]);
+
+        expect(results[0].asArray()).toEqual([
+            {
+                name: 'bob',
+                age: 122,
+                _key: makeKey([122, 'bob'], 'md5')
+            },
+            {
+                name: 'joe',
+                age: 34,
+                _key: makeKey([34, 'joe'], 'md5')
+            },
+            {
+                name: 'frank',
+                age: 99,
+                _key: makeKey([99, 'frank'], 'md5')
+            }
+        ]);
+    });
+});
+
+function makeKey(values: unknown[], algo: string) {
```

Comment:
I see what you're staying, it's the same code.   I wanted a something outside of the processor to verify the key is correct and built with the proper configs.  Otherwise if the it's wrong in the processor how would I know?

Reply:
I can get the correct key string and use that.

Reply:
removed duplicate function from tests